src/material-experimental/mdc-slider/slider-adapter.ts

```diff
@@ -6,133 +6,179 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
+import {ChangeDetectorRef, ElementRef, Inject} from '@angular/core';
 import {SpecificEventListener, EventType} from '@material/base';
 import {MDCSliderAdapter, Thumb, TickMark} from '@material/slider';
+import {MatSliderThumb, MAT_SLIDER} from './slider-thumb';
 
+/**
+ * This is a dummy interface that just contains the properties and methods of MatSlider that are
+ * used by SliderAdapter. Rather than directly referencing MatSlider, we use this interface when
+ * to avoid a circular dependency between MatSlider and SliderAdapter.
+ */
+interface MatSlider {
```

Comment:
🤮

Reply:
I would rename this interface to something else, perhaps `SliderThumbContainer`, and ensure that it's not exported through the public API. Also, as discussed earlier, there should only need to be one.